great_expectations/datasource/fluent/sources.py

```diff
@@ -458,7 +463,7 @@ def add_datasource(
                     datasource_type, name_or_datasource, **kwargs
                 )
             ) or (
-                datasource_type(name=name_or_datasource, **kwargs)
+                datasource_type(name=name_or_datasource, **kwargs)  # type: ignore[arg-type] # could be Datasource - expect str
```

Comment:
I think this will always be a string since in the first part of the `or`, the method  call `self._datasource_passed_in(...)` will return a datasource if one was passed in. You can leave this `ignore` but could update the reason.